stm32-modules/include/thermocycler-refresh/thermocycler-refresh/motor_task.hpp

```diff
@@ -0,0 +1,178 @@
+        : message_queue(q),
+          task_registry(nullptr),
+          lid_stepper_state{.status = LidStepperState::IDLE} {}
+    MotorTask(const MotorTask& other) = delete;
+    auto operator=(const MotorTask& other) -> MotorTask& = delete;
+    MotorTask(MotorTask&& other) noexcept = delete;
+    auto operator=(MotorTask&& other) noexcept -> MotorTask& = delete;
+    ~MotorTask() = default;
+    auto get_message_queue() -> Queue& { return message_queue; }
+    void provide_tasks(tasks::Tasks<QueueImpl>* other_tasks) {
+        task_registry = other_tasks;
+    }
+
+    template <typename Policy>
+    requires MotorExecutionPolicy<Policy>
+    auto run_once(Policy& policy) -> void {
+        auto message = Message(std::monostate());
+
+        // This is the call down to the provided queue. It will block for
+        // anywhere up to the provided timeout, which drives the controller
+        // frequency.
+
+        static_cast<void>(message_queue.recv(&message));
+        std::visit(
+            [this, &policy](const auto& msg) -> void {
+                this->visit_message(msg, policy);
+            },
+            message);
+    }
+
+  private:
+    template <typename Policy>
+    auto visit_message(const std::monostate& _ignore, Policy& policy) -> void {
+        static_cast<void>(_ignore);
+        static_cast<void>(policy);
+    }
+
+    template <typename Policy>
+    auto visit_message(const messages::LidStepperDebugMessage& msg,
+                       Policy& policy) -> void {
+        // check for errors
+        auto error = errors::ErrorCode::NO_ERROR;
+        if (lid_stepper_state.status == LidStepperState::MOVING) {
+            error = errors::ErrorCode::LID_MOTOR_BUSY;
+        } else if (policy.lid_stepper_check_fault()) {
+            error = errors::ErrorCode::LID_MOTOR_FAULT;
+        }
+        if (error == errors::ErrorCode::NO_ERROR) {
+            // Start movement and cache the id for later
+            policy.lid_stepper_set_dac(motor_util::LidStepper::current_to_dac(
+                LID_STEPPER_DEFAULT_CURRENT));
+            policy.lid_stepper_start(
+                motor_util::LidStepper::angle_to_microsteps(msg.angle));
+            lid_stepper_state.status = LidStepperState::MOVING;
+            lid_stepper_state.response_id = msg.id;
+        } else {
+            auto response = messages::AcknowledgePrevious{
+                .responding_to_id = msg.id, .with_error = error};
+            static_cast<void>(
+                task_registry->comms->get_message_queue().try_send(
+                    messages::HostCommsMessage(response)));
+        }
+    }
+
+    template <typename Policy>
+    auto visit_message(const messages::LidStepperComplete& msg, Policy& policy)
+        -> void {
+        static_cast<void>(msg);
+        if (lid_stepper_state.status == LidStepperState::MOVING) {
+            lid_stepper_state.status = LidStepperState::IDLE;
+            policy.lid_stepper_set_dac(0);
```

Comment:
do we want to bind this in here? you're the expert but i'm a bit surprised we want to turn off the motor when done moving, as opposed to maintaining some amount of holding current. I guess in the fully-open or fully-closed case this is right, but maybe it's better if that's a separate message if we end up needing to either change that or string moves together.